### PR TITLE
fragments are fixed and working dynamically

### PR DESCRIPTION
**Pull request by c2khanna**
made fragments dynamic. tested and works fine. please review

### CODE REVIEW

File: app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java [hunk 1]
```diff
@@ -118,26 +136,19 @@ public void onSubmitOrderClick(View view) {
...
-                    MenuItem currentItem = temp.adapter.getItem(j);
-                    if(currentItem.isCheck){
-                        selectedItems.add(currentItem);
-                    }
-                }
-            }else if(currentFragment instanceof MainCourseFragment){
-                MainCourseFragment temp = (MainCourseFragment) currentFragment;
-                for(int j = 0 ; j < temp.adapter.getCount(); j++){
-                    MenuItem currentItem = temp.adapter.getItem(j);
+            GenericFragment currentFragment = (GenericFragment) adapter.getFragment(i);
```
Comment (andrewmhc) [inline app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java]: are we going to be using a generic fragment for all of our activities? how adaptable is it?

File: app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java [hunk 2]
```diff
@@ -65,22 +71,34 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        for (int i = 0 ; i < drinksNames.length ; i++){
+            listOfDrinks.add(new MenuItem(drinksNames[i], false));
+            drinksJSONArray.put(listOfDrinks.get(i).getJSONObject());
+        }
+
+        JSONArray dessertsJSONArray = new JSONArray();
+        for(int i = 0; i < dessertsNames.length; i++){
+            listOfDesserts.add(new MenuItem(dessertsNames[i], false));
+            dessertsJSONArray.put(listOfDesserts.get(i).getJSONObject());
+        }
```
Comment (andrewmhc) [inline app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java:63]: maybe can refactor to a generic method that adds MenuItem objects to a list

**Review by andrewmhc — APPROVED** [review]
looks good, maybe consider adding high level comments for the files; might not be needed considering the code is quite readable

File: app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java (hunk 2 above)
Comment (c2khanna) [reply app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java:63]: waiting on the format of the response to be finalized and it coming from the backend. so once we have that we can refactor it

File: app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java (hunk 1 above)
Comment (c2khanna) [reply app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java]: at least for the menu page this is a generic fragment. I will actually rename it to genericMenuFragment or something

**Pull request merged**